overlays/ghc-packages.nix

```diff
@@ -29,12 +29,52 @@ let
       cabal-to-nix *.cabal > $out
     '';
   };
-  importCabal = name: src:
+
+  combineFiles = name: ext: files: self.stdenv.mkDerivation {
+    inherit name;
+    phases = [ "buildPhase" ];
+
+    buildPhase = ''
+      mkdir $out
+    '' + self.lib.concatStrings (self.lib.mapAttrsToList (name: file: ''
+      cp ${file} $out/${name}${ext}
+    '') files);
+  };
+
```

Comment:
combineFiles is needed to avoid having one nix derivation per ghc-boot-package (less fixed output hashes to maintain).

Reply:
Can we just use `symlinkJoin` or something?

Reply:
I thought about `symlinkJoin`, but I think it might only work if we changed the cabal2Nix derivation to make a directory output instead of a single file.  I'd rather keep that bit simple in case one day we get to rip out all the materialization code.  I was also not sure if symlinks show up differently from files in the nix sha256 (I don't think it follows them right or it would be horribly slow?) and since we can't have symlinks to the store in the materialized version we would still need another derivation to `cp -r` them to make them into files.

Reply:
Maybe `linkFarm`?

Reply:
Cool.  Had not seen `linkFarm`.  I have changed the code to use that and updated `materialize.nix` so that it always runs `cp -r` on the derivation (links will be turned into files).